xbmc/input/InputCodingTableBaiduPY.cpp

```diff
@@ -121,6 +122,30 @@ std::vector<std::wstring> CInputCodingTableBaiduPY::GetResponse(int response)
   return words;
 }
 
+bool CInputCodingTableBaiduPY::Initialize()
+{
+  CSingleLock lock(m_CS);

```

Comment:
CThread has its own mutex. Maybe CThread::m_CriticalSection should be locked in Create()